### PR TITLE
feature/updates-to-dotnet-6

### PR DESCRIPTION
**Pull request by dandev486**
## changelog

⬆️ updates project files to dotnet 6 and csharp 10
🔧 fixes dotnet versions on github action 
🔧 includes changes from @leandromoh original attempt #40 
🔥 removes operating system matrix (was building only on ubuntu-latest)

### CODE REVIEW

**Comment by dandev486** [comment]
## Test Evidence

🟢 github action is working


🟢 project is being tested against multiple framework versions

File: .github/workflows/test-dotnet.yml [hunk 1]
```diff
@@ -4,18 +4,18 @@ on: [ pull_request, push ]
...
-        operating-system: [ubuntu-latest ]
-        dotnet-version: ['5.0']
-    name: Dotnet ${{ matrix.dotnet }} sample
+    name: dotnet test
     steps:
       - uses: actions/checkout@v2
       - name: Setup dotnet
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: ${{ matrix.dotnet }}
```
Comment (dandev486) [inline .github/workflows/test-dotnet.yml:19]: by the way, even though this line is no longer necessary, it should actually be using the same name as defined at L12.

```
dotnet-version: ${{ matrix.dotnet-version }}
```

File: .github/workflows/test-dotnet.yml [hunk 2]
```diff
@@ -4,18 +4,18 @@ on: [ pull_request, push ]
...
-      - run: dotnet build RecordParser.Test/RecordParser.Test.csproj
```
Comment (dandev486) [inline .github/workflows/test-dotnet.yml:20]: and no need to build prior to testing, the `dotnet test` command already restores and builds the project implicitly!

File: .github/workflows/test-dotnet.yml (hunk 1 above)
Comment (dandev486) [reply .github/workflows/test-dotnet.yml:19]: Ref.: as requested, [docs](https://github.com/actions/setup-dotnet)

**Comment by leandromoh** [comment]
Hi @dandev486 thanks for your contribution! 
Could you rebase with master and resolve the conflict please?

**Comment by dandev486** [comment]
> Hi @dandev486 thanks for your contribution! Could you rebase with master and resolve the conflict please?

Since the fork was already deleted, this was done on #47, closing this one!

**Pull request closed without merging**